src/Datadog.Trace/Agent/ZipkinApi.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Agent
+{
+    internal class ZipkinApi : IApi
+    {
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<ZipkinApi>();
+
+        private readonly TracerSettings _settings;
+        private Uri _tracesEndpoint;
+
+        public ZipkinApi(TracerSettings settings)
+        {
+            Log.Debug("Creating new Zipkin Api");
+
+            _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+            _tracesEndpoint = _settings.AgentUri; // User needs to include the proper path.
+        }
+
+        public void SetBaseEndpoint(Uri baseEndpoint)
+        {
+            // TODO: This doesn't make sense for this exporter.
+            _tracesEndpoint = new Uri(baseEndpoint, _tracesEndpoint.PathAndQuery);
+        }
+
+        public async Task<bool> SendTracesAsync(Span[][] traces)
+        {
+            if (traces == null || traces.Length == 0)
+            {
+                // Nothing to send, no ping for Zipkin.
+                return true;
+            }
+
+            // retry up to 5 times with exponential back-off
+            var retryLimit = 5;
+            var retryCount = 1;
+            var sleepDuration = 100; // in milliseconds
+
+            while (true)
+            {
+                // TODO: Initially same code for Fx and Core.
+                var request = WebRequest.CreateHttp(_tracesEndpoint);
+                request.Method = "POST";
+                request.ContentType = "application/json";
+
+                foreach (var header in _settings.HeaderTags)
```

Comment:
I'm surprised to see this here and I think this is not meant to be used here. We use `_settings.HeaderTags` only to decorate web server spans based on the request headers. We don't send these headers to our agent

Reply:
Thanks for noticing that. I completely confused this with some other exporter (OTel collector?) that has a similarly named configuration for the function above. I will remove it.